auth/api/iam/metadata.go

```diff
@@ -22,32 +22,73 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/nuts-foundation/go-did/did"
 	"github.com/nuts-foundation/nuts-node/auth/oauth"
 	"github.com/nuts-foundation/nuts-node/core"
 	"github.com/nuts-foundation/nuts-node/crypto/jwx"
+	"github.com/nuts-foundation/nuts-node/vdr/didweb"
 )
 
-func authorizationServerMetadata(identity url.URL, oauth2BaseURL url.URL) oauth.AuthorizationServerMetadata {
+func authorizationServerMetadata(ownedDID did.DID) (*oauth.AuthorizationServerMetadata, error) {
+	if ownedDID.Method == "web" {
+		return _authzMetadataDidWeb(ownedDID)
+	}
+	return _authzMetadataBase(ownedDID), nil
+}
+
+func _authzMetadataDidWeb(ownedDID did.DID) (*oauth.AuthorizationServerMetadata, error) {
```

Comment:
weird underscore prefix?

Reply:
having an underscore doesn't prevent this, just makes the code weirder. Either comment and accept it's not 100% fool proof, or move to a seperate package to improve isolation. Or just merge the 3 into 1 function with the `did:web` functionality in an `if`

Reply:
merged